### ISSUE TITLE
@todo Marker not being reported when inside a comment block

### ISSUE THREAD

**Issue by blackdrago** [phpDocumentor/phpDocumentor2#1842]
I'm using phpdocumentor 2.9 (though phpdoc --version says 2.8.5?). I've added a number of TODO markers to my project's files. None of them were appearing, despite my xml config file including the TODO marker.

So I went through and changed the marker in a few places, trying things like @todo, TODO, todo: and so on, wondering if I had simply failed to use the correct marker. Not a single one worked. 

Then I tried the following:
// @todo message

And it suddenly worked. It appears that phpdocumentor will not parse @todo (no matter how you format them) if said marker is inside a comment block tag.  I don't understand the validity of this choice, since I will often add @todo to a function descriptor's comment block when I plan to make future changes overall (as opposed to a single, inline comment which doesn't make a lot of sense if the todo applies to the function as a whole, rather than  a line/portion of it).

According to your site's own documentation/tutorial, phpdocumentor should parse @todo when inside a comment block. In fact, the only example on the @todo page uses it like this.

URL: https://www.phpdoc.org/docs/latest/references/phpdoc/tags/todo.html
 /**
  * Counts the number of items in the provided array.
  *
  * @todo add an array parameter to count
  *
  * @return int Returns the number of elements.
  */
 function count()
 {
     <...>
 }

This is why I believe this is a bug (@todo marker not being parsed/handled inside of comment tags) rather than a documentation error... that, and I don't see anything anywhere that implies phpdocumentor will only accept markers on inline comment tags.

**Comment by jaapio** [comment]
Thanks for your report.

**Comment by blackdrago** [comment]
Update: I've discovered that part of my initial report was wrong. 

I during my testing, I added @todo to the list of markers in my phpdoc.dist.xml file. It looked something like this:
```
<phpdoc>
    <markers>
        <item>TODO</item>
        <item>todo</item>
        <item>@todo</item>
    </markers>
...
</phpdoc>
```
This is enabled the follownig line to be picked up by phpdoc
// @todo something important 

However, negative side effect: most of the templates (any with a structure.xml, believe), would fail with an encoding error.   I traced it to the following file/line:

/usr/share/pear/phpDocumentor/src/phpDocumentor/Plugin/Core/Transformer/Writer/Xml.php 
Line 218
$marker_obj = new \DOMElement(strtolower($marker['type']));
DOMElements can't start with a number or other special characters (like @), hence the failure...

I've removed the problematic marker, and now none of my todos are being found anywhere, since they're all // @todo whatever

I changed a few of them to // todo whatever and // TODO something and those are being found. 

TODO and todo inside of comment blocks are still not appearing, and the @todo tag still isn't being added to any documentation.

**Comment by mvriel** [comment]
Hello @blackdrago,

You are right that by default phpDocumentor does not recognize `@todo` as an inline marker (`// @todo something`) and adding that to your configuration as a recognized marker is the correct course of action.

It is unfortunate that this causes issues with XML based templates and for that I have created a fix in commit a27da12bbfc13c8c01c713e5fd9fbf5f3bae14bc. This is now available on master and will be included in the next release.

Thank you for making us aware of this issue
